libnavigation-core/src/main/java/com/mapbox/navigation/core/routealternatives/AlternativeRouteProgressDataProvider.kt

```diff
@@ -0,0 +1,37 @@
+package com.mapbox.navigation.core.routealternatives
+
+import com.mapbox.navigation.core.RouteProgressData
+
+internal object AlternativeRouteProgressDataProvider {
+
+    fun getRouteProgressData(
+        primaryRouteProgressData: RouteProgressData,
+        alternativeMetadata: AlternativeRouteMetadata
+    ): RouteProgressData {
+        val legIndex: Int
+        val routeGeometryIndex: Int
+        val legGeometryIndex: Int?
+        val primaryFork = alternativeMetadata.forkIntersectionOfPrimaryRoute
+        val alternativeFork = alternativeMetadata.forkIntersectionOfAlternativeRoute
+        val isForkPointAheadOnPrimaryRoute =
+            primaryRouteProgressData.routeGeometryIndex < primaryFork.geometryIndexInRoute
+        if (isForkPointAheadOnPrimaryRoute) {
+            val legIndexDiff = primaryFork.legIndex - alternativeFork.legIndex
+            legIndex = primaryRouteProgressData.legIndex - legIndexDiff
+            val routeGeometryIndexDiff =
+                primaryFork.geometryIndexInRoute - alternativeFork.geometryIndexInRoute
+            routeGeometryIndex =
+                primaryRouteProgressData.routeGeometryIndex - routeGeometryIndexDiff
+            val legGeometryIndexDiff =
+                primaryFork.geometryIndexInLeg - alternativeFork.geometryIndexInLeg
+            legGeometryIndex = primaryRouteProgressData.legGeometryIndex?.let {
+                it - legGeometryIndexDiff
+            }
```

Comment:
```suggestion
            val distanceToFork = primaryFork.geometryIndexInRoute - primaryRouteProgressData.routeGeometryIndex
            routeGeometryIndex = alternativeFork.geometryIndexInRoute - distanceToFork
            legGeometryIndex = alternativeFork.geometryIndexInLeg - distanceToFork
```
this would be much more readable to me

Reply:
Well, I don't know, I find my approach more readable. :) I think it depends on the way you're thinking about it, and I was thinking exactly the way the code is written.
What's more, it uses the approach that's used for calculating legIndex.
It looks like it's subjective, so I'll leave it as is unless someone else thinks the way with `distanceToFork` is better.

Reply:
I'm proposing a change because it took me an unnecessary long time and napkin backtesting to understand what the code tries to do and if it's even correct :upside_down_face: What throws me off is that we calculate diff of indices that are not clearly related to each other (indices of two completely different routes). I mean this:
> primaryFork.geometryIndexInRoute - alternativeFork.geometryIndexInRoute

Without fully understanding the context this calculation is very surprising.

The proposal is subjectively more readable but also has one less calculation and avoids null-checks so I'd prefer us merging it unless it's functionally incorrect.

Reply:
Honestly, if we can do the same simplification for leg index calculation, I'd love us doing that too :)

Reply:
Either way, the proposed code suffers from https://github.com/mapbox/mapbox-navigation-android/pull/6848#discussion_r1081257785 as well (partially). The calculations for `legGeometryIndex` is only correct if the for point is on the current leg.

Reply:
Let's close this thread since the proposal is functionally incorrect. Let's see what a correct implementation would look like and discuss there.